pkg/phlaredb/symdb/stacktrace_tree.go

```diff
@@ -153,10 +153,10 @@ func newParentPointerTree(size uint32) *parentPointerTree {
 }
 
 func (t *parentPointerTree) resolve(dst []int32, id uint32) []int32 {
+	dst = dst[:0]
```

Comment:
Apart from that It might also make sense to empty slices before putting them into the pool:

https://github.com/grafana/pyroscope/blob/b6816e31eb0c8418e74d9f7d57783fb983cdeb89/pkg/phlaredb/symdb/partition_memory.go#L205

Reply:
I've been thinking about this, but in practice, it doesn't improve anything: the slice is reused many times before it is put back. So I decided to keep it as is. To be honest, the pool is barely helpful; I should have removed it